deploy/200-role.yaml

```diff
@@ -3,18 +3,8 @@ kind: ClusterRole
 metadata:
   name: shipwright-build-controller
 rules:
-- apiGroups:
-  - ""
-  resources:
-  - pods
-  - services
-  - services/finalizers
-  - endpoints
-  - persistentvolumeclaims
-  - events
-  - configmaps
-  - secrets
-  - serviceaccounts
+- apiGroups: ['']
+  resources: ['secrets', 'serviceaccounts']
```

Comment:
This looks correct for service accounts. Secrets, I do not think we create or update or delete them. And I think we removed all code that lists secrets. So, imo we currently need:

- Secrets: `get`, `watch`
- ServiceAccounts: `create`, `delete`, `update`, `get`

Reply:
Thanks, that's helpful. Except for SA create and update, this also matches the behavior observed using [rbac-audit](https://github.com/imjasonh/rbac-audit), which is good signal.

This might mean we don't have an e2e test covering SA creation, which is maybe fine since we expect to remove it anyway.

I've manually added SA create+update to the generated policy (which I've manually cleaned up for readability).